src/Inventory/Request.php

```diff
@@ -349,18 +373,121 @@ public function inventory($data)
     }
 
    /**
-    * Handle agent inventory task request
+    * Handle agent enabled inventory task support on contact request
+    *
+    * @param array $params Required hooks params
     *
     * @return array
     */
-    public function handleInventoryTask()
+    public function handleInventoryTask(array $params): array
     {
+       // Preset response as glpi supports native inventory by default
+        $params['options']['response'][self::INVENT_TASK] = [
+            'server' => 'glpi',
+            'version' => GLPI_VERSION
+        ];
+        $params = Plugin::doHookFunction(Hooks::HANDLE_INVENTORY_TASK, $params);
 
-        $params['options']['response'] = [];
-        $params['item'] = $this->inventory->getAgent();
-        $params = Plugin::doHookFunction("handle_inventory_task", $params);
+       // Return inventory task support
+        return $params['options']['response'][self::INVENT_TASK] ?? [];
```

Comment:
Before this change, result was extracted from `$params['options']['response']`, and now it is extracted from `$params['options']['response'][self::INVENT_TASK]`. I guess this change is necessary. In this case, all plugins that are using this hook must be updated.

Reply:
@tsmr Does ocsinventoryng uses this hook ?

Reply:
Afaik, this is a new hook related to json protocol CONTACT request. The only plugin I'm aware of is databaseinventory plugin which will insert some infos to tell agent it has databases credentials the agent can request later. I checked it and the infos are effectively returned under `$params['options']['response'][self::INVENT_TASK]` so the plugin is still compatible with this change.
I think it's better to do this way to not permit a hook to change any information related to another task. This is a little more secure IMHO.